addons/portal/wizard/portal_invite_share.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo import api, fields, models, _
+
+
+class PortalInviteShare(models.TransientModel):
+    _name = 'portal.invite.share'
+
+    @api.model
+    def default_get(self, fields):
+        result = super(PortalInviteShare, self).default_get(fields)
+        result['res_model'] = self._context.get('active_model')
+        result['res_id'] = self._context.get('active_id')
+        result['share_link'] = self.env[result['res_model']].browse(result['res_id'])._get_share_url()
+        return result
+
+    res_model = fields.Char('Related Document Model', required=True)
+    res_id = fields.Integer('Related Document ID', required=True)
+    partner_ids = fields.Many2many('res.partner', string="Recipients", required=True)
+    note = fields.Text(help="Add extra content to display in the email")
+    share_link = fields.Char(string="Link", compute='_compute_share_link')
+    access_warning = fields.Text("Access warning", compute="_compute_share_link")
+
+    @api.depends('res_model', 'res_id')
+    def _compute_share_link(self):
+        base_url = self.env['ir.config_parameter'].sudo().get_param('web.base.url')
+        for rec in self:
+            res_model = self.env[rec.res_model]
+            if isinstance(res_model, self.pool['portal.mixin']):
+                record = res_model.browse(rec.res_id)
+                rec.access_warning = record.access_warning
+                rec.share_link = base_url + record._get_share_url()
```

Comment:
Share link computed here is not the same as the one computed in default. Actually a computed field does not require a default value normally as it will be computed based on its trigger.

I also thinks you don't have to compute the whole URL based on base_url. Normally urls are made global in message_post (replace_local_links). Could you have a look ?

Reply:
base url is needed to display the generic url in the share wizard.